data/lib/core/player.lua

```diff
@@ -769,3 +769,79 @@ do
 		return lossPercents[blessingCount]
 	end
 end
+
+function Player.getMaxTrackedBestiary(self)
+	return 50
+end
+
+function Player.setTrackedBestiary(self, raceId, checked)
+	local trackedCount = self:getTrackedBestiaryCount()
+	if checked and trackedCount >= self:getMaxTrackedBestiary() then
+		self:sendTextMessage(MESSAGE_STATUS_WARNING, "You have reached the maximum number of trackable creatures.\nYou have to remove one of your currently tracked creatures before you can add another one.")
+		return false
+	end
+
+	self:setStorageValue(PlayerStorageKeys.bestiaryTrackerBase + raceId, checked and 1 or -1)
+
+	local trackedBestiary = Game.getTrackedBestiary()[self:getId()] or {}
+	local index = table.indexOf(trackedBestiary, raceId)
+
+	if checked and not index then
+		table.insert(trackedBestiary, raceId)
+	elseif not checked and index then
+		table.remove(trackedBestiary, index)
+	end
```

Comment:
I was thinking in having a big sparse table and operating like this:


```suggestion
    trackedBestiary[raceId] = checked or nil
```

(`or nil` to remove it instead of just setting to false)

Can we rely directly on storage values instead, and drop this table?